sdk/python/feast/client.py

```diff
@@ -977,197 +601,126 @@ def ingest(
-                store must be a historical store.
-            features: Optional list of feature names to filter from the results.
-            ingestion_ids: Optional list of dataset Ids by which to filter data
-                before retrieving featureStatistics. Cannot be used with start_date
-                and end_date.
-                If multiple dataset ids are provided, unaggregatable featureStatistics
-                will be dropped.
-            start_date: Optional start date over which to filter statistical data.
-                Data from this date will be included.
-                Cannot be used with dataset_ids. If the provided period spans
-                multiple days, unaggregatable featureStatistics will be dropped.
-            end_date: Optional end date over which to filter statistical data.
-                Data from this data will not be included.
-                Cannot be used with dataset_ids. If the provided period spans
-                multiple days, unaggregatable featureStatistics will be dropped.
-            force_refresh: Setting this flag to true will force a recalculation
-                of featureStatistics and overwrite results currently in the cache, if any.
-            project: Manual override for default project.
+        batch_source_type = SourceType(feature_table.batch_source.type).name
 
-        Returns:
-           Returns a tensorflow DatasetFeatureStatisticsList containing TFDV featureStatistics.
-        """
+        try:
+            if batch_source_type == "BATCH_FILE":
+                from urllib.parse import urlparse
 
-        if ingestion_ids is not None and (
-            start_date is not None or end_date is not None
-        ):
-            raise ValueError(
-                "Only one of dataset_id or [start_date, end_date] can be provided."
-            )
+                file_url = feature_table.batch_source.file_options.file_url[:-1]
+                uri = urlparse(file_url)
+                staging_client = get_staging_client(uri.scheme)
 
-        if project != "" and "/" not in feature_set_id:
-            feature_set_id = f"{project}/{feature_set_id}"
+                file_name = dest_path.split("/")[-1]
+                date_today = datetime.datetime.today().strftime("%Y-%m-%d")
 
-        request = GetFeatureStatisticsRequest(
-            feature_set_id=feature_set_id,
-            features=features,
-            store=store,
-            force_refresh=force_refresh,
-        )
-        if ingestion_ids is not None:
-            request.ingestion_ids.extend(ingestion_ids)
-        else:
-            if start_date is not None:
-                request.start_date.CopyFrom(
-                    Timestamp(seconds=int(start_date.timestamp()))
+                staging_client.upload_file(
+                    dest_path,
+                    uri.hostname,
+                    str(uri.path).strip("/") + "/" + f"date={date_today}/" + file_name,
```

Comment:
we should partitioned by `date(event_timestamp)` instead of today.
`eventTimestamp` and `datePartitioned` columns are specified in FeatureTable.
With this partitioning it could several directories that we need to upload.

Reply:
Updated.